test/metrics/query.go

```diff
@@ -0,0 +1,260 @@
+func PodExecCreate(client *kubernetes.Clientset, name, cmName string, labels map[string]string) error {
+	execPod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: name,
+			Labels:    labels,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "test",
+					Image: "centos:7",
+					Command: []string{
+						"sh",
+						"-c",
+						"trap exit TERM; while true; do sleep 5; done",
+					},
+					Env: []corev1.EnvVar{
+						{
+							// This allows secure curl command
+							Name:  "CURL_CA_BUNDLE",
+							Value: "/var/certs/service-ca.crt",
+						},
+					},
+				},
+			},
+		},
+	}
+	cm, err := client.CoreV1().ConfigMaps(name).Get(cmName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	volumes := execPod.Spec.Volumes
+	volumeMounts := execPod.Spec.Containers[0].VolumeMounts
+	volumes, volumeMounts = addCMMount(volumes, volumeMounts, cm)
+
+	execPod.Spec.Volumes = volumes
+	execPod.Spec.Containers[0].VolumeMounts = volumeMounts
+	execPod, err = client.CoreV1().Pods(name).Create(execPod)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// addCMMount is forked from service-ca-operator/test/e2e/e2e_test.go
+func addCMMount(volumes []corev1.Volume, volumeMounts []corev1.VolumeMount, cm *corev1.ConfigMap) ([]corev1.Volume, []corev1.VolumeMount) {
+	var volumesDefaultMode int32 = 420
+	volumes = append(volumes, corev1.Volume{
+		Name: cm.ObjectMeta.Name,
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{Name: cm.ObjectMeta.Name},
+				DefaultMode:          &volumesDefaultMode,
+			},
+		},
+	})
+
+	volumeMounts = append(volumeMounts, corev1.VolumeMount{
+		Name:      cm.ObjectMeta.Name,
+		MountPath: "/var/certs",
+		ReadOnly:  true,
+	})
+
+	return volumes, volumeMounts
+}
+
+// Forked from test/e2e/framework so as not to vendor kubernetes/test/e2e/framework
+// kubectlBuilder is used to build, customize and execute a kubectl Command.
+type kubectlBuilder struct {
+	cmd     *exec.Cmd
+	timeout <-chan time.Time
+}
+
+// newKubectlCommand returns a kubectlBuilder for running kubectl.
+func newKubectlCommand(kubeconfig, kubectl string, args ...string) *kubectlBuilder {
+	b := new(kubectlBuilder)
+	b.cmd = kubectlCmd(kubeconfig, kubectl, args...)
+	return b
+}
+
+// kubectlCmd runs the kubectl executable through the wrapper script.
+func kubectlCmd(kubeconfig, kubectl string, args ...string) *exec.Cmd {
+	defaultArgs := []string{}
+	defaultArgs = append(defaultArgs, "--kubeconfig="+kubeconfig)
+	kubectlArgs := append(defaultArgs, args...)
+	cmd := exec.Command(kubectl, kubectlArgs...)
+	return cmd
+}
+
+// runExecCmd runs the given cmd in the context of the given pod using `kubectl exec`
+// inside of a shell.
+func runExecCmd(kubeconfig, kubectl, name, cmd string) (string, error) {
```

Comment:
There's a lot of these functions, squash to a single one? If we need to we'll extract helpers as we go.

Reply:
ok, will do

Reply:
no longer required w/ querying against external url, no exec required.